kata-deploy/scripts/kata-deploy.sh

```diff
@@ -88,74 +88,28 @@ function configure_crio() {
 	cp -n "$crio_conf_file" "$crio_conf_file_backup"
 
+	local kata_root="/run/vc"
+
 	local kata_clh_path="/opt/kata/bin/kata-clh"
-	local kata_clh_conf="crio.runtime.runtimes.kata-clh"
+	local kata_clh_conf="kata-clh"
 
 	local kata_fc_path="/opt/kata/bin/kata-fc"
-	local kata_fc_conf="crio.runtime.runtimes.kata-fc"
+	local kata_fc_conf="kata-fc"
 
 	local kata_qemu_path="/opt/kata/bin/kata-qemu"
-	local kata_qemu_conf="crio.runtime.runtimes.kata-qemu"
+	local kata_qemu_conf="kata-qemu"
 
 	local kata_qemu_virtiofs_path="/opt/kata/bin/kata-qemu-virtiofs"
-	local kata_qemu_virtiofs_conf="crio.runtime.runtimes.kata-qemu-virtiofs"
-
-	# add kata-qemu config
-	if grep -q "\[$kata_qemu_conf\]" $crio_conf_file; then
-		echo "Configuration exists $kata_qemu_conf, overwriting"
-		sed -i "/\[$kata_qemu_conf\]/,+1s#runtime_path.*#runtime_path = \"${kata_qemu_path}\"#" $crio_conf_file
-	else
-		cat <<EOT | tee -a "$crio_conf_file"
-
-# Path to the Kata Containers runtime binary that uses the QEMU hypervisor.
-[$kata_qemu_conf]
-  runtime_path = "${kata_qemu_path}"
-EOT
-	fi
-
-        # add kata-qemu-virtiofs config
-	if grep -q "\[$kata_qemu_virtiofs_conf\]" $crio_conf_file; then
-		echo "Configuration exists $kata_qemu_virtiofs_conf, overwriting"
-		sed -i "/\[$kata_qemu_virtiofs_conf\]/,+1s#runtime_path.*#runtime_path = \"${kata_qemu_virtiofs_path}\"#" $crio_conf_file
-	else
-		cat <<EOT | tee -a "$crio_conf_file"
-
-# Path to the Kata Containers runtime binary that uses the QEMU hypervisor with virtiofs support.
-[$kata_qemu_virtiofs_conf]
-  runtime_path = "${kata_qemu_virtiofs_path}"
-EOT
-        fi
-
-	# add kata-fc config
-	if grep -q "\[$kata_fc_conf\]" $crio_conf_file; then
-		echo "Configuration exists for $kata_fc_conf, overwriting"
-		sed -i "/\[$kata_fc_conf\]/,+1s#runtime_path.*#runtime_path = \"${kata_fc_path}\"#" $crio_conf_file
-	else
-		cat <<EOT | tee -a "$crio_conf_file"
-
-# Path to the Kata Containers runtime binary that uses the firecracker hypervisor.
-[$kata_fc_conf]
-  runtime_path = "${kata_fc_path}"
-EOT
-	fi
-
-	# add kata-clh config
-	if grep -q "\[$kata_clh_conf\]" $crio_conf_file; then
-		echo "Configuration exists $kata_clh_conf, overwriting"
-		sed -i "/\[$kata_clh_conf\]/,+1s#runtime_path.*#runtime_path = \"${kata_clh_path}\"#" $crio_conf_file
-	else
-		cat <<EOT | tee -a "$crio_conf_file"
-
-# Path to the Kata Containers runtime binary that uses the Cloud Hypervisor.
-[$kata_clh_conf]
-  runtime_path = "${kata_clh_path}"
-EOT
-	fi
-
-	# Replace if exists, insert otherwise
-	grep -Fq 'manage_network_ns_lifecycle =' $crio_conf_file \
-		&& sed -i '/manage_network_ns_lifecycle =/c manage_network_ns_lifecycle = true' $crio_conf_file \
-		|| sed -i '/\[crio.runtime\]/a manage_network_ns_lifecycle = true' $crio_conf_file
+	local kata_qemu_virtiofs_conf="kata-qemu-virtiofs"
+
+	# add kata configurations
+	crio --config "$crio_conf_file" \
+		--runtimes "${kata_clh_conf}:${kata_clh_path}:${kata_root}" \
```

Comment:
@amshinde yeah, that is the advantage of it, but had to remove it because this runs inside a container, which would not have the `crio` binary installed.